src/web/src/app/auth/service.ts

```diff
@@ -121,29 +146,31 @@ export class AuthService {
 	}
 
 	goToLogin(): void {
+		this.stopRefreshTimer();
+		localStorage.removeItem('first');
+		this._token = null;
+		this._isAuthorized.next(false);
 		window.location.href = window.location.origin + this._apiUrl + "login";
 	}
 
 	goToLogout(): void {
-		clearTimeout(this._timeoutHandle);
-		localStorage.removeItem('first');
-		this._isAuthorized.next(false);
-
 		if (!this._token || this.isTokenExpired(this._token)) {
 			this.goToLogin();
 			return;
 		}
 
-		this._token = null;
-		window.location.href = window.location.origin + this._apiUrl + 'logout';
+		this.stopRefreshTimer();
+		window.location.href = window.location.origin + this._apiUrl + 'logout?token=' + this._token;
 	}
 
-	logout() {
-		// remove user from local storage to log user out
-		localStorage.removeItem('first');
-		if (this._token) {
-			this._token = null;
-			clearTimeout(this._timeoutHandle);
-		}
+	getUserEmailById(userId: string) {
+		let userAddInfo = this._mockDataForSpaceDetailPage.find(u => u.userId === userId);
+		return userAddInfo?.email;
+	}
+
+	getUserNameById(userId: string) {
+		let userAddInfo = this._mockDataForSpaceDetailPage.find(u => u.userId === userId);
+		return userAddInfo?.name;
 	}
```

Comment:
Это должно быть в src/app/user/service.ts
```suggestion
	getById(userId: string): Observable<User> {
		const user = this._mockDataForSpaceDetailPage.find(u => u.userId === userId);
		return of(user);
	}
```

Reply:
Готово.